src/main/connect/exchange_msg.rs

```diff
@@ -90,15 +95,22 @@ impl<UID: Uid> ExchangeMsg<UID> {
 
     fn receive_response(&mut self, core: &mut Core, poll: &Poll) {
         match self.socket.read::<Message<UID>>() {
-            Ok(Some(Message::Connect(their_uid, name_hash))) => {
+            Ok(Some(Message::Connect(their_uid, name_hash, _their_pk))) => {
                 if their_uid != self.expected_id || name_hash != self.expected_nh {
                     return self.handle_error(core, poll);
                 }
                 let _ = core.remove_state(self.token);
                 let token = self.token;
-                let socket = mem::replace(&mut self.socket, Default::default());
 
-                (*self.finish)(core, poll, token, Some(socket));
+                let mut socket = mem::replace(&mut self.socket, Default::default());
+                if socket
```

Comment:
It might be better to use `match` in these cases to log errors (in `debug!()`) to help debuggability in case something goes wrong.